tapdb/assets_store.go

```diff
@@ -379,6 +380,8 @@ type AssetStore struct {
 	clock clock.Clock
 
 	txHeights *lru.Cache[chainhash.Hash, cacheableBlockHeight]
+
+	sync.Mutex
```

Comment:
Should give this mutex a name, so it's more clear where it's used. For example: `proofInsertMtx`?